hazelcast/src/main/java/com/hazelcast/spi/impl/DelegatingMigrationAwareService.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl;
+
+import com.hazelcast.spi.MigrationAwareService;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.PartitionMigrationEvent;
+import com.hazelcast.spi.PartitionReplicationEvent;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A {@link MigrationAwareService} that delegates to another {@link MigrationAwareService} and keeps track of the number of
+ * migrations concerning the primary replica (either as current or new replica index) currently in-flight.
+ */
+public class DelegatingMigrationAwareService implements MigrationAwareService {
+
+    private static final int PRIMARY_REPLICA_INDEX = 0;
+
+    private final MigrationAwareService migrationAwareService;
+    private final AtomicInteger migrationsInFlight;
```

Comment:
minor comment: can you add a comment or rename the field to specify it tracks only the partition owner migrations?